### PR TITLE
Fix Flaky Discover Table Tests

### PR DESCRIPTION
**Pull request by LDrago27**
### Description

Fix Flaky tests in Discover Table.

Without Security

https://github.com/opensearch-project/opensearch-dashboards-functional-test/assets/38322563/e800b3fd-989e-444e-b5e4-eec8b1e62c1f

With Security

https://github.com/opensearch-project/opensearch-dashboards-functional-test/assets/38322563/bc377c53-ce91-4e42-903c-4bbb2e34bdcf

### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js [hunk 1]
```diff
@@ -459,9 +464,9 @@ describe('discover_table', () => {
           .then((noEntries) => {
             cy.setTopNavDate(
               'Sep 22, 2015 @ 14:00:00.000',
-              'Sep 22, 2015 @ 18:00:00.000'
+              'Sep 22, 2015 @ 14:30:00.000'
             );
-            cy.waitForLoader();
```
Comment (ashwin-pc) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js:464]: why did you change this? wait for loader is more reliable since it waits for a specific element and not just the an arbitrary time which can cause the test to become flakey

Comment (LDrago27) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js:464]: While running the tests, I found that waiting for page load didn't make sense in few occasions. While applying the filter to a table the entire page doesn't reload. In those scenarios instead of waiting for the number of entires to change , cypress picks up the older result and the evaluation condition fails. 
To avoid such scenarios I have put a wait time.

Comment (LDrago27) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js:464]: @ashwin-pc Updated the PR to remove waits, instead added verification for hits which has resolved the previous issues.

**Review by ashwin-pc — APPROVED** [review]

**Review by ananzh — APPROVED** [review]
